test/extensions/filters/http/jwt_authn/jwt_cache_test.cc

```diff
@@ -0,0 +1,65 @@
+#include <memory>
+
+#include "envoy/extensions/filters/http/jwt_authn/v3/config.pb.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/jwt_authn/jwt_cache.h"
+
+#include "test/extensions/filters/http/jwt_authn/test_common.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+using envoy::extensions::filters::http::jwt_authn::v3::JwtProvider;
+using ::google::jwt_verify::Status;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+class JwtCacheTest : public testing::Test {
+protected:
+  const std::string config = R"({
+      enable_jwt_cache: true,
+      jwt_cache_size: 20
+  })";
+  void SetUp() override {
+    jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
+    Status status = jwt_->parseFromString(GoodToken);
+    if (status == Status::Ok) {
+      setupCache(config);
+    }
+  }
+
+  void setupCache(const std::string& config_str) {
+    TestUtility::loadFromYaml(config_str, provider_);
+    cache_ =
+        JwtCache::create(provider_.enable_jwt_cache(), provider_.jwt_cache_size(), time_system_);
```

Comment:
This unit-test is just for JwtCache.   No need to start the test from "Provider".    Just use JwtCache::create directly and test its cache code

Reply:
I am fetching the JwtProvider config, So started from Provider.